usr/src/uts/common/fs/zfs/sys/spa.h

```diff
@@ -606,6 +606,28 @@ typedef enum spa_import_type {
 	SPA_IMPORT_ASSEMBLE
 } spa_import_type_t;
 
+/*
+ * Should we force sending TRIM commands even to devices which evidently
+ * don't support it?
+ *	OFF: no, only send to devices which indicated support
+ *	ON: yes, force send to everybody
+ */
+typedef enum {
+	SPA_FORCE_TRIM_OFF = 0,	/* default */
+	SPA_FORCE_TRIM_ON
+} spa_force_trim_t;
+
+/*
+ * Should we send TRIM commands in-line during normal pool operation while
+ * deleting stuff?
+ *	OFF: no
+ *	ON: yes
+ */
+typedef enum {
+	SPA_AUTO_TRIM_OFF = 0,	/* default */
+	SPA_AUTO_TRIM_ON
+} spa_auto_trim_t;
```

Comment:
What does `auto_trim=on` do if none of the disks claim to support TRIM?  Same goes for `zpool trim`.

What impact is there of sending the TRIM command to a device that doesn't claim to support it?

Rather than separate `force_trim` and `auto_trim` properties, I think it would be easier to understand as:
`auto_trim=off|auto|force` plus the `zpool trim` command could have a `--force` flag.